### ISSUE TITLE
heart-beat header

### ISSUE THREAD

**Issue by dimonik** [NaikSoftware/StompProtocolAndroid#18]
Does your library support heart-beat mechanism?

According to spec:

> The heart-beat header is OPTIONAL. A missing heart-beat header MUST be treated the same way as a “heart-beat:0,0” header, that is: the party cannot send and does not want to receive heart-beats.

**Comment by NaikSoftware** [comment]
no.. always send 0,0

1 нояб. 2016 г. 1:47 ПП пользователь "dimonik" <notifications@github.com>
написал:

**Comment by Prakash199** [comment]
Does this library supports ping pong feature because the connection is no more alive after fer minutes

**Comment by steviemo** [comment]
Yes we are seeing this too - the stomp client disconnects after maybe 1 or 2 mins. Is there a workaround for this?

**Comment by steviemo** [comment]
OK turns out the STOMP client is being closed due to ELB - not by our spring server. A direct connection to our spring server using this client maintains a connection. I think the ELB is closing as connection is idle and this doesn't happen with using stomp.js as it has a heart beat so keeps the WS connection active.

I'll look at implementing a heartbeat for this library that can be enabled/disabled and will submit a pull request

**Comment by steviemo** [comment]
For other that might use this thread I was able to implement my own heart beat thread which starts in the OPENED lifecycle event and stops on ERROR or CLOSED:

`private void startHeartbeat() {
        heartbeat = heartbeatExecutor.scheduleAtFixedRate(new Runnable() {
            @Override
            public void run() {
                Log.e(TAG, "SEND HEARTBEAT FROM SERVICE");
                if (null != stompClient) {
                    stompClient.send("", "h").subscribe();
                }
            }
        }, 0, 25, TimeUnit.SECONDS);
    }

    private void stopHeartbeat() {
        if (null != heartbeat) {
            heartbeat.cancel(true);
            heartbeat = null;
        }
    }`
